go/enclave/gas/oracle.go

```diff
@@ -40,25 +46,36 @@ func (o *oracle) EstimateL1CostForMsg(args *gethapi.TransactionArgs, block *type
 		encoded = append(encoded, *args.Data...)
 	}
 
-	return o.calculateL1Cost(block, encoded, header)
+	return o.calculateL1Cost(block, header, encoded)
 }
 
-func (o *oracle) calculateL1Cost(block *types.Header, encodedTx []byte, header *common.BatchHeader) (*big.Int, error) {
-	// If the block does not have excess blob gas, we can't estimate the cost
-	if block.ExcessBlobGas == nil {
-		return big.NewInt(0), nil
+// calculateL1Cost - Calculates the L1 cost as a multiple of the L2 base fee.
+func (o *oracle) calculateL1Cost(l1Block *types.Header, l2Batch *common.BatchHeader, encodedTx []byte) (*big.Int, error) {
+	totalCost := big.NewInt(0)
+
+	// If the l1Block does not have excess blob gas, we can't estimate the cost
+	if l1Block.ExcessBlobGas == nil {
+		return totalCost, nil
 	}
 
-	blobFee := eip4844.CalcBlobFee(*block.ExcessBlobGas)
+	// price in Wei for a single unit of blob
+	blobFeePerByte := eip4844.CalcBlobFee(*l1Block.ExcessBlobGas)
+
+	// Calculate teh cost of including the tx in a blob
+	txL1Size := CalculateL1Size(encodedTx)
+	shareOfBlobCost := big.NewInt(0).Mul(txL1Size, blobFeePerByte)
+
+	// Add a value to the shareOfBlobCost to account for the cost of the transaction itself
+	shareOfL1TxGas := big.NewInt(L1TxGas / TxsPerBatch)
```

Comment:
I dont understand this - we have blob fee per byte and we multiply it by the bytes of the transaction and this is the final cost. why is the rest of the stuff needed, it feels wrong.

Reply:
oh its the execution cost of the l1 transaction